openedx/frontend-app-library-authoring
# feat: View/Edit tags for Library Components

## Description

This PR adds the ability to view/edit tags for component blocks of V2 Libraries. This utilizes the already implemented content tags drawer in the [course authoring MFE](https://github.com/openedx/frontend-app-course-authoring/) repo, rendering it inside an iframe.

If blocks have tags applied, the count is shown and can be clicked to open the manage tags drawer. Otherwise the manage tags option is available in the actions dropdown menu.

<img width="931" alt="Screen Shot 2023-12-29 at 4 07 20 PM" src="https://github.com/openedx/frontend-app-library-authoring/assets/6829768/3383a0da-8108-45d8-aa3f-789020795e57">

## Supporting Information
Related Ticket: 
- Resolves https://github.com/openedx/modular-learning/issues/121
- Depends on https://github.com/openedx/frontend-app-course-authoring/pull/771
- Depends on https://github.com/openedx/edx-platform/pull/33979

## Testing Instructions

1. Checkout https://github.com/openedx/frontend-app-course-authoring/pull/771 and run it locally
2. Checkout https://github.com/openedx/edx-platform/pull/33979 and run it locally
3. Run this PR locally
4. Make sure you have sample taxonomy data locally, check out https://github.com/open-craft/taxonomy-sample-data/ if needed
5. Navigate to http://localhost:3001/ and create a new V2 Library if it doesnt already exist
6. Navigate to the newly created Library and add a few blocks/components to it
7. Click on the 3-dotted more actions button in one of the blocks/components, and select `Mange tags`, the side drawer should open and load taxonomy tags to choose from
8. Try adding/removing tags then close the block
9. Refresh the page, you should notice a button with currently applied tags count on it, clicking on it should also open up the side drawer. Note: The count currently only shows applied tags count, not including the implicit tags. The implicit tags count implementation in the latest openedx-learning version that has not been released yet.

---
Private-ref: [FAL-3599](https://tasks.opencraft.com/browse/FAL-3599)
